tests/test_persistence.py

```diff
@@ -1852,6 +1852,23 @@ def next2(update, context):
         assert nested_ch.conversations[nested_ch._get_key(update)] == 1
         assert nested_ch.conversations == pickle_persistence.conversations['name3']
 
+    @pytest.mark.parametrize(
+        'filepath',
+        ['pickletest', Path('pickletest')],
+        ids=['str filepath', 'pathlib.Path filepath'],
+    )
+    def test_filepath_argument_types(self, filepath):
+        pick_persist = PicklePersistence(
+            filepath=filepath,
+            single_file=False,
+            on_flush=False,
+        )
+        pick_persist.update_user_data(1, 1)
+
+        assert pick_persist.get_user_data()[1] == 1
+        filepath = Path(filepath)
+        assert filepath.with_name(filepath.name + "_user_data").is_file()
```

Comment:
```suggestion
        pick_persist = PicklePersistence(
            filepath=filepath,
            on_flush=False,
        )
        pick_persist.update_user_data(1, 1)

        assert pick_persist.get_user_data()[1] == 1
        assert Path(filepath).is_file()
```
just a bit shorter :)

Reply:
but then the test will fail.
in ```PicklePersistence.update_user_data``` there is this line:
```self._dump_file(Path(f"{self.filepath}_user_data"), self.user_data)```

so in this test i need to add the "_user_data" to the end of the path, hence the long line.